packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -30,6 +30,25 @@ import 'table.dart';
 /// See also the discussion at [WidgetsBinding.drawFrame].
 bool debugPrintRebuildDirtyWidgets = false;
 
+/// Signature for [debugOnRebuildDirtyWidget] implementations.
```

Comment:
@Hixie The most controversial part of the CL is adding the two debugging hooks
debugOnRebuildDirtyWidget and debugOnProfilePaint.
What I would like is a way for the inspector to be notified over each widget rebuild and render object repaint.
This seems like the most minimal option to support this use case in debug builds but I'm open to other options.

Reply:
Seems fine to me, the way you added it is very clean. I actually thought the most controversial part was the addition of more hooks in widgets/binding.dart that defer to the WidgetInspectorService singleton. :-)

Reply:
thanks for the feedback!